byte_mlperf/backends/MIGRAPHX/compile_backend_migraphx.py

```diff
@@ -0,0 +1,153 @@
+import os
+import json
+import logging
+import copy
+
+import torch
+# import lego
+
+import tensorflow as tf
+
+import onnx
+
+import migraphx
+
+tf.get_logger().setLevel('ERROR')
+
+import numpy as np
+import time
+from byte_mlperf.backends import compile_backend
+import subprocess
+
+log = logging.getLogger("CompileBackendMIGRAPHX")
+from byte_mlperf.tools import saved_to_onnx, torch_to_onnx
+
+pt_dtype_map = {
+    "FLOAT32": torch.float32,
+    "FLOAT16": torch.float16,
+    "INT8": torch.int8,
+    "INT32": torch.int32,
+    "LONG": torch.long,
+    "BOOL": torch.bool
+}
+
+tf_dtype_map = {
+    "FLOAT32": tf.float32,
+    "FLOAT16": tf.float16,
+    "INT32": tf.int32,
+}
+
+INPUT_TYPE = {
+    "UINT8": np.uint8,
+    "FLOAT32": np.float32,
+    "LONG": np.long,
+    "INT32": np.int32,
+    "INT64": np.int64,
+    "BOOL": np.bool,
+}
+
+
+class CompileBackendMIGRAPHX(compile_backend.CompileBackend):
+    def __init__(self):
+        super(CompileBackendMIGRAPHX, self).__init__()
+        self.target = 'rocm'
+        self.hardware_type= 'MIGRAPHX'
+
+    def compile(self, config, dataloader=None):
+        self.model_name = config['model_info']['model']
+        self.framework = config['model_info']['framework']
+        self.input_type = config['model_info']['input_type'].split(",")
+
+        model_paths = config['model_info']['model_path'].split('.')
+        model_path = config['model_info']['model_path']
+        test_data = dataloader.get_fake_samples(
+            1, config['model_info']['input_shape'], self.input_type)
+        input_tensors = []
+        #convert tf/pytorch model to onnx model, tf works, not verify pytorch now.
+        model_onnx_path=model_path
+        if self.framework == "Tensorflow":
+            model_onnx_path = os.path.join(model_path,self.model_name+ '.onnx')
+            if not os.path.exists(model_onnx_path):
+                saved_to_onnx.savedmodel_to_onnx(model_path, model_onnx_path)
+        elif self.framework == "Pytorch":
+            raise NotImplementedError("MIGraphX backend for models of PyTorch framework has not been implemented yet.")
```

Comment:
I see the file imported torch_to_onnx but didn't use it here. Is there any reason?

Reply:
Have fixed it.